apps/api/src/database/migrations/1730724383210-ArchiveCompletedNotifications.ts

```diff
@@ -0,0 +1,166 @@
+import { MigrationInterface, QueryRunner, Table, TableForeignKey } from 'typeorm';
+
+export class ArchiveCompletedNotifications1730724383210 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Drop foreign key from notify_notification_retries
+    const table = await queryRunner.getTable('notify_notification_retries');
+    const foreignKey = table.foreignKeys.find(
+      (fk) => fk.columnNames.indexOf('notification_id') !== -1,
+    );
+    await queryRunner.dropForeignKey('notify_notification_retries', foreignKey);
+
+    // Create table archived_notifications
+    await queryRunner.createTable(
+      new Table({
+        name: 'archived_notifications',
+        columns: [
+          {
+            name: 'id',
+            type: 'int',
+            isPrimary: true,
+            isGenerated: true,
+            generationStrategy: 'increment',
+          },
+          {
+            name: 'notification_id',
+            type: 'int',
+          },
+          {
+            name: 'channel_type',
+            type: 'tinyint',
+          },
+          {
+            name: 'data',
+            type: 'text',
+            isNullable: true,
+          },
+          {
+            name: 'delivery_status',
+            type: 'tinyint',
+            default: 1,
+          },
+          {
+            name: 'result',
+            type: 'text',
+            isNullable: true,
+          },
+          {
+            name: 'created_on',
+            type: 'timestamp',
+            default: 'CURRENT_TIMESTAMP',
+          },
+          {
+            name: 'updated_on',
+            type: 'timestamp',
+            default: 'CURRENT_TIMESTAMP',
+            onUpdate: 'CURRENT_TIMESTAMP',
+          },
+          {
+            name: 'created_by',
+            type: 'varchar',
+          },
+          {
+            name: 'updated_by',
+            type: 'varchar',
+          },
+          {
+            name: 'status',
+            type: 'tinyint',
+            default: 1,
+          },
+          {
+            name: 'application_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'provider_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'retry_count',
+            type: 'int',
+            default: 0,
+          },
+        ],
+      }),
+    );
```

Comment:
I was not sure about 1 initially so did not add it, maybe can add now

For 2, in the original notification entity we did not add a character limit so copied that over to archived notifications, there is a possibility of data not being populated if one column has restriction and other does not

For 3, ChannelType and DeliveryStatus max limit can increase if we decide to add new providers, Maybe we can do `constants.ChannelType.length` as upper limit. We already build query that fetches data based on DeliveryStatus so it may be redundant

Reply:
Okay just do 1 if that is required.

Reply:
Add indexes for notify_notifications and notify_archived_notifications 25df0d16071269b1349b18b9a81a64ee7222dd21